apiserver/facades/client/application/application_unit_test.go

```diff
@@ -1146,6 +1146,28 @@ func (s *ApplicationSuite) TestDeployCAASModelCharmNeedsNoOperatorStorage(c *gc.
 	c.Assert(result.Results[0].Error, gc.IsNil)
 }
 
+func (s *ApplicationSuite) TestDeployCAASModelSidecarCharmNeedsNoOperatorStorage(c *gc.C) {
+	s.model.modelType = state.ModelTypeCAAS
+	delete(s.model.cfg, "operator-storage")
+	s.backend.charm = &mockCharm{
+		meta:     &charm.Meta{},
+		manifest: &charm.Manifest{Bases: []charm.Base{{}}},
```

Comment:
I'm kind of surprised we just needed a Bases declaration and not a Containers declaration. Because v2 machine charm would look like this.

Reply:
The format v2 check is (currently) premised on having bases defined. It's irrelevant how v2 is determined. All we care about is whether the charm is v2 or not. No v2 charm requires operator storage so we just use that to short circuit the check.